sanic_ext/extensions/injection/injector.py

```diff
@@ -15,6 +15,32 @@
 def add_injection(app: Sanic, injection_registry: InjectionRegistry) -> None:
     signature_registry = _setup_signature_registry(app, injection_registry)
 
+    @app.before_server_start
+    def register_needed_signals(app: Sanic, _):
+        app.signal_router.reset()
+
+        for signal in injection_registry.signals:
+
+            @app.signal(signal)
+            async def inject_kwargs(request, route, kwargs, **_):
```

Comment:
These args are not going to be the same. I would do:

```python
async def inject_kwargs(request, **_):
```

And then extract everything from the `Request` object.